packages/hasura-storage-js/tests/uploadFile.test.ts

```diff
@@ -0,0 +1,73 @@
+import faker from '@faker-js/faker'
+import { interpret } from 'xstate'
+import { waitFor } from 'xstate/lib/waitFor'
+import { createFileUploadMachine } from '../src/machines'
+import { Typegen0 } from '../src/machines/file-upload.typegen'
+import { BASE_URL } from './helpers/config'
+import { uploadFileInternalErrorHandler, uploadFiledNetworkErrorHandler } from './helpers/handlers'
+import server from './helpers/server'
+import { clientStorage } from './helpers/storage'
+import { GeneralFileUploadState } from './helpers/types'
+import { createAuthMachine } from '@nhost/core'
+
+type FileUploadState = GeneralFileUploadState<Typegen0>
+
+// TODO complete tests when MSW supports upload events
+// ! https://github.com/mswjs/interceptors/issues/187
+
+const auth = interpret(
+  createAuthMachine({
+    backendUrl: BASE_URL + '/v1/auth',
+    clientUrl: 'http://localhost:3000',
+    autoRefreshToken: false,
+    autoSignIn: false,
+    clientStorageType: 'custom',
+    clientStorage
+  })
+)
+
+const fileUploadMachine = createFileUploadMachine({
+  url: BASE_URL + '/v1/storage',
+  auth
+})
+
+const fileUploadService = interpret(fileUploadMachine)
+
+beforeAll(async () => {
+  server.listen({ onUnhandledRequest: 'error' })
+  await waitFor(auth.start(), (state) => state.matches('authentication.signedIn'))
+})
+afterAll(() => server.close())
+
+beforeEach(() => {
+  fileUploadService.start()
+})
+
+afterEach(() => {
+  fileUploadService.stop()
+  server.resetHandlers()
+})
+
+test(`should fail if there is a network error`, async () => {
+  server.use(uploadFiledNetworkErrorHandler)
+
+  const file = new File([], 'test.txt')
+  fileUploadService.send({
+    type: 'UPLOAD',
+    file,
+    id: undefined,
+    bucketId: undefined,
+    name: undefined
+  })
+  const state: FileUploadState = await waitFor(fileUploadService, (state: FileUploadState) =>
+    state.matches('error')
+  )
+  console.log(state.context)
```

Comment:
We miss the assertion here. Could you please also remove this `console.log` statement?

Reply:
yes. deactivated test. I could not find a way to make MSW work with file upload events:
https://github.com/mswjs/interceptors/issues/187

Reply:
Could you please change this to `test.skip` then?